### PR TITLE
[FIX] l10n_ca: saskatchewan tax report

### PR DESCRIPTION
**Pull request by guva-odoo**
Saskatchewan tax report was missing a line for
`Total before Sales`

opw-3955926

### CODE REVIEW

File: addons/l10n_ca/data/template/account.tax-ca_2023.csv [hunk 1]
```diff
@@ -7,7 +7,7 @@
 "","","","","","","","","","","","tax","invoice","+Ca103||+CaQc103","l10n_ca_231000","",""
 "","","","","","","","","","","","base","refund","-Ca101||-CaBcA||-CaQc101","","",""
 "","","","","","","","","","","","tax","refund","-Ca103||-CaQc103","l10n_ca_231000","",""
-"pst_sale_tax_6","6% PST SK","6%","6% PST SK","sale","6.0","percent","20","False","tax_group_pst_6","","base","invoice","","","6% TVP SK","6% TVP SK"
+"pst_sale_tax_6","6% PST SK","6%","6% PST SK","sale","6.0","percent","20","False","tax_group_pst_6","","base","invoice","+CaSk8_base","","6% TVP SK","6% TVP SK"
 "","","","","","","","","","","","tax","invoice","+CaSk8","l10n_ca_232000","",""
 "","","","","","","","","","","","base","refund","","","",""
```
Comment (JulienAlardot) [inline addons/l10n_ca/data/template/account.tax-ca_2023.csv:12]:
```suggestion
"","","","","","","","","","","","base","refund","-CaSk8Base","","",""
```
Confirmed with Laura that we need to decrease it if we refund

**Review by JulienAlardot — CHANGES_REQUESTED** [review]
:+1:

**Review by JulienAlardot — APPROVED** [review]
@antoine162 LGTM

**Review by antoine162 — COMMENTED** [review]
From what I understand, there are 2 tax report forms for Saskatchewan, which can be found on their [gov's website](https://www.saskatchewan.ca/business/taxes-licensing-and-reporting/provincial-taxes-policies-and-bulletins/provincial-sales-tax/file-your-pst-return):
- the [PST return](https://publications.saskatchewan.ca/api/v1/products/214/formats/214/download)
- the [PST worksheet supplement](https://publications.saskatchewan.ca/api/v1/products/23529/formats/30306/download)

I think the customer wants to see the 'PST return' in Odoo. But, from what I can see, our tax report for Saskatchewan is the 'PST worksheet supplement'. (looking at the l10n_ca task, this also appears to be what we implemented for Saskatchewan)

Unfortunately, looking closely at the two forms, it seems to me that the 'PST worksheet supplement' is not a more detailed version of the PST return. Indeed the 'PST worksheet supplement' seems to be exclusively related to computing PST credits. 

This is also how the PST return refers to the PST worksheet supplement:
image

Note that the PST worksheet supplement doesn't include a box for the total sales, since it's a really specialized sheet just for computing tax credits. From what I can see, the version we have in Odoo already corresponds to the official form and doesn't really needs a fix.

So basically, I think the problem here is that we didn't implement the main PST return for Saskatchewan, and probably the solution is to have a dev task to implement it.

**Comment by JulienAlardot** [comment]
> From what I understand, there are 2 tax report forms for Saskatchewan, which can be found on their [gov's website](https://www.saskatchewan.ca/business/taxes-licensing-and-reporting/provincial-taxes-policies-and-bulletins/provincial-sales-tax/file-your-pst-return):

There are 2 return reports, the government GST declaration and the PST return. But there is a worksheet to help you fill the PST return, which is what we implemented at first

> 
> * the [PST return](https://publications.saskatchewan.ca/api/v1/products/214/formats/214/download)
> * the [PST worksheet supplement](https://publications.saskatchewan.ca/api/v1/products/23529/formats/30306/download)
> 
> I think the customer wants to see the 'PST return' in Odoo. But, from what I can see, our tax report for Saskatchewan is the 'PST worksheet supplement'. (looking at the l10n_ca task, this also appears to be what we implemented for Saskatchewan)
> 
> Unfortunately, looking closely at the two forms, it seems to me that the 'PST worksheet supplement' is not a more detailed version of the PST return. Indeed the 'PST worksheet supplement' seems to be exclusively related to computing PST credits.
It is:
image
image

> 
> This is also how the PST return refers to the PST worksheet supplement: image
> 
> Note that the PST worksheet supplement doesn't include a box for the total sales, since it's a really specialized sheet just for computing tax credits. From what I can see, the version we have in Odoo already corresponds to the official form and doesn't really needs a fix.
It kinda does, we are not implementing the form "ready to send" as it needs to be their form and we don't want to implement it. They can copy the amounts manually 

> 
> So basically, I think the problem here is that we didn't implement the main PST return for Saskatchewan, and probably the solution is to have a dev task to implement it.

@antoine162 This has been seen with Laura as to which solution to implement. The Worksheet helps with the computation of all the fields required for the normal PST return and was the thing I implemented at first. It was indeed only lacking one field to allow the report being useable. As their form has a specific format they'll have to do it "manually" anyway

File: addons/l10n_ca/data/template/account.tax-ca_2023.csv [hunk 2]
```diff
@@ -7,9 +7,9 @@
 "","","","","","","","","","","","tax","invoice","+Ca103||+CaQc103","l10n_ca_231000","",""
 "","","","","","","","","","","","base","refund","-Ca101||-CaBcA||-CaQc101","","",""
 "","","","","","","","","","","","tax","refund","-Ca103||-CaQc103","l10n_ca_231000","",""
-"pst_sale_tax_6","6% PST SK","6%","6% PST SK","sale","6.0","percent","20","False","tax_group_pst_6","","base","invoice","","","6% TVP SK","6% TVP SK"
+"pst_sale_tax_6","6% PST SK","6%","6% PST SK","sale","6.0","percent","20","False","tax_group_pst_6","","base","invoice","+CaSk8Base","","6% TVP SK","6% TVP SK"
 "","","","","","","","","","","","tax","invoice","+CaSk8","l10n_ca_232000","",""
-"","","","","","","","","","","","base","refund","","","",""
+"","","","","","","","","","","","base","refund","-CaSk8Base","","",""
```
Comment (antoine162) [inline addons/l10n_ca/data/template/account.tax-ca_2023.csv:12]:
```suggestion
"","","","","","","","","","","","base","refund","","","",""
```

Since the worksheet supplement counts credit notes under tax credits, Laura and I think the total sales should not include refunds.

image

**Review by antoine162 — COMMENTED** [review]
Okay thank you so much, it all makes sense and I confirmed with Laura that we want to insert the Total Sales line at this place :smile: 

Just one comment regarding the changes to the tax, see below.

File: addons/l10n_ca/data/template/account.tax-ca_2023.csv (hunk 2 above)
Comment (JulienAlardot) [reply addons/l10n_ca/data/template/account.tax-ca_2023.csv:12]: ... She told me the opposite xD
And the worksheet contains what is collected, what is refunded and they actually report the difference on the final report (where the total sales are inputed). So the total sales and the total taxes would not match :thinking:

File: addons/l10n_ca/data/tax_report.xml [hunk 3]
```diff
@@ -668,178 +668,211 @@
...
+                    </record>
+                </field>
+            </record>
+            <record id="l10n_ca_tr_pst_sk_step_2" model="account.report.line">
+                <field name="name">Step 2</field>
+                <field name="code">CaSkStep2</field>
+                <field name="hierarchy_level">0</field>
+                <field name="children_ids">
+                    <record id="l10n_ca_tr_pst_sk_total_sales" model="account.report.line">
+                        <field name="name">Total Sales Before Taxes</field>
```
Comment (antoine162) [inline addons/l10n_ca/data/tax_report.xml:756]: Sorry, another change request :joy: we discussed this with Laura a bit further, and in the end we would like this line to be at the top of the report (before 'Step 1'). It's just that since we're not counting refunds under the total sales, it makes sense to put the total sales before all the computations of tax credits. Thanks!

**Review by antoine162 — APPROVED** [review]
Thank you so much!
@robodoo r+
